### PR TITLE
Makefile

### PR DESCRIPTION
**Pull request by rhatdan**
These two patches shrink the size of the binaries that we are shipping.

First patch eliminates ostree from all of the executables that do not support it.  It also adds scripts to determine if SELinux and OSTree are supported on the platform.

The second patch adds the -s and -w flags which strip the binraries giving you abour 40% less size.

### CODE REVIEW

**Comment by rhatdan** [comment]
This PR is in response to https://github.com/kubernetes-incubator/cri-o/issues/1034

**Comment by rhatdan** [comment]
@mrunalp @runcom PTAL

**Comment by rhatdan** [comment]
@nalind @mitrmac PTAL

File: Makefile [hunk 1]
```diff
@@ -72,19 +73,19 @@ pause:
...
 crio: .gopathok $(shell hack/find-godeps.sh $(GOPKGDIR) cmd/crio $(PROJECT))
-	$(GO) build $(LDFLAGS) -tags "$(BUILDTAGS)" -o $@ $(PROJECT)/cmd/crio
+	$(GO) build $(LDFLAGS) -tags "$(BUILDTAGS) containers_image_ostree_stub" -o $@ $(PROJECT)/cmd/crio
 
 crioctl: .gopathok $(shell hack/find-godeps.sh $(GOPKGDIR) cmd/crioctl $(PROJECT))
-	$(GO) build $(LDFLAGS) -tags "$(BUILDTAGS)" -o $@ $(PROJECT)/cmd/crioctl
+	$(GO) build $(LDFLAGS) -tags "$(BUILDTAGS) containers_image_ostree_stub" -o $@ $(PROJECT)/cmd/crioctl
 
 kpod: .gopathok $(shell hack/find-godeps.sh $(GOPKGDIR) cmd/kpod $(PROJECT))
 	$(GO) build $(LDFLAGS) -tags "$(BUILDTAGS)" -o $@ $(PROJECT)/cmd/kpod
```
Comment (nalind) [inline Makefile]: Shouldn't `kpod` and `crio` have the same build tags, so that they get the same driver support?

Comment (mheon) [reply Makefile]: Agree, kpod should have the same flags as CRI-O. We're expecting them to share storage, so this seems like a good idea.

Comment (rhatdan) [reply Makefile]: kpod can push to an ostree repo.

**Comment by mtrmac** [comment]
ACK on `hack/ostree_tag.sh`.

**Comment by rhatdan** [comment]
@mrunalp @runcom @nalind WDYT Can I merge this?

**Pull request merged**